src/adapters/EventAdapters.ts

```diff
@@ -0,0 +1,92 @@
+import { EventEmitter } from "events"
+import type { createClient } from "redis"
+import postgres from "postgres"
+
+export type EventsAdapterEmitFn = (event: string, data: string) => Promise<void>
+export type EventsAdapterOnFn = (event: string, fn: (data: string, event: string) => void) => Promise<void>
+export type EventsAdapterInitFn = () => Promise<void>
+
+export class EventsAdapter {
```

Comment:
Create a specific file for each adapter

- EventAdapter
- EmitterEventsAdapter
- RedisEventsAdapter
- Postgres

Each file has to `export default` the correct class